### ISSUE TITLE
Justifed Nav within Navbar smashes the NavItems together

### ISSUE THREAD

**Issue by mtscout6** [react-bootstrap/react-bootstrap#1272]
When I tried putting `justified` on a `Nav` within a `Navbar` the output looks smashed.

<img width="710" alt="screen shot 2015-09-01 at 9 10 12 pm" src="https://cloud.githubusercontent.com/assets/736870/9622267/ea78eabe-50ed-11e5-8604-afec762407f1.png">

```
const navbarInstance = (
  <Navbar brand={<a href="#">React-Bootstrap</a>}>
    <Nav justified>
      <NavItem eventKey={1} href='#'>Link</NavItem>
      <NavItem eventKey={2} href='#'>Link</NavItem>
      <NavDropdown eventKey={3} title='Dropdown' id='nav-brand-dropdown'>
        <MenuItem eventKey='1'>Action</MenuItem>
        <MenuItem eventKey='2'>Another action</MenuItem>
        <MenuItem eventKey='3'>Something else here</MenuItem>
        <MenuItem divider />
        <MenuItem eventKey='4'>Separated link</MenuItem>
      </NavDropdown>
    </Nav>
  </Navbar>
);

React.render(navbarInstance, mountNode);
```

Remove the `Navbar` wrapper and all is fine.

**Comment by kwnccc** [comment]
Hi @mtscout6 , do you think I can help with this bug? I have worked before with Navbrand trying to componetize the brand attribute to NavBrand. So if there is not anyone else going to work on this one I'd like to contribute.

**Comment by kwnccc** [comment]
I was having a look in bootstrap documentation here: http://getbootstrap.com/components/#navbar-default There is a note: "Justified navbar nav links are currently not supported."

But there is an example here with justified nav items: http://getbootstrap.com/examples/justified-nav/

I have concluded that when 'nav-justified' is applied then 'navbar-nav' should not be added because it overrides the nav's li floating.

**Comment by mtscout6** [comment]
That example is precisely the one I was trying to implement for a demo I'm working on! Thanks for digging into this. I was seeing the side effects of what you are mentioning, though I didn't have the time to tackle it last night. In fact I'm moving on to a different example for my demo due to time constraints.

I did try manually removing the `navbar-nav` class as you suggest but it is still off:

![screen shot 2015-09-02 at 8 06 36 am](https://cloud.githubusercontent.com/assets/736870/9633345/a174e4d8-5149-11e5-9aa4-1e6498adfeee.png)

**Comment by kwnccc** [comment]
For this purpose we could some added styles to match the same design, but only for the docs needs. Just like customizing/overriding bootstrap theme on a demo.

**Comment by mtscout6** [comment]
I was hoping to simply state in my presentation that you can easily implement any of the Bootstrap examples fairly easily with React-Bootstrap, though this is not proving to be the case even for another example. See: #1277.

I'd like to avoid adding custom styling here, instead I'd rather beg the question about what needs to change within our React components so they play nice with expected Bootstrap behavior.

I wonder if this is a case covered by [Bootlint](http://www.bootlint.com/) if so then we should think about this

**Comment by kwnccc** [comment]
I totally agree with your state, but this issue also the case for the Boostrap behavior. If you inspect the styles of their example http://getbootstrap.com/examples/justified-nav/ you can see that they also have added their own custom styles for the demo purposes. There is an extra <code>justified-nav.css</code> file which is not part of the main Bootstrap library.

![screen shot 2015-09-03 at 08 41 26](https://cloud.githubusercontent.com/assets/1969742/9651310/b2530ed2-5217-11e5-92e6-caf981f0e8fc.png)
![screen shot 2015-09-03 at 08 41 49](https://cloud.githubusercontent.com/assets/1969742/9651309/b25313e6-5217-11e5-9928-2c9ba3a4c629.png)

**Comment by taion** [comment]
Closed by #1419 for the time being given the absence of upstream support.
